core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -111,7 +116,12 @@ public AtomicFileWriter(@Nonnull Path destinationPath, @Nonnull Charset charset)
         }
 
         try {
-            tmpPath = Files.createTempFile(dir, "atomic", "tmp");
+            Set<PosixFilePermission> permissions = Jenkins.DEFAULT_FILE_PERMISSIONS;
+            if (Files.exists(destinationPath)) {
```

Comment:
Do not bother doing this.
AtomicFileWriter has no way to specify previously and would have kept the OS default.
If anything then ATF API should be expanded to take a permission set but this is out of scope.

Also this will fail to run on some file systems causing exceptions that are not caught to always be thrown - and hence no file could ever be written